src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -906,6 +907,81 @@ static bool usesReceiver(BoundExpression argument)
             }
         }
 
+        [return: NotNullIfNotNull(nameof(receiver))]
+        BoundExpression? AdjustReceiverForExtensionsIfNeeded(BoundExpression? receiver, Symbol member)
+        {
+            if (receiver is null)
+            {
+                return receiver;
+            }
+
+            Debug.Assert(member is not null);
+            if (receiver is BoundTypeExpression)
+            {
+                return receiver;
+            }
+
+            if (receiver is BoundThisReference)
+            {
+                // PROTOTYPE this was only minimally tested to observe side-effects but needs more work (in CheckValue for instance)
+                // In an extension type, `this.UnderlyingTypeMember` is replaced with `this.<UnderlyingInstanceField>$.UnderlyingTypeMember`
+                Debug.Assert(receiver.Type is not null);
+                if (receiver.Type.IsExtension && !member.ContainingType.IsExtension)
+                {
+                    var thisExtensionType = (SourceExtensionTypeSymbol)receiver.Type;
+                    Debug.Assert(thisExtensionType.UnderlyingInstanceField is not null);
+                    Debug.Assert(thisExtensionType.UnderlyingInstanceField.Type.Equals(member.ContainingType, TypeCompareKind.AllIgnoreOptions));
+
+                    return _factory.Field(receiver, thisExtensionType.UnderlyingInstanceField);
```

Comment:
>return _factory.Field(receiver, thisExtensionType.UnderlyingInstanceField); [](http://example.com/codeflow?start=20&length=75)

It is not obvious when this transformation should happen when interpolation handlers are involved. A PROTOTYPE comment should be good enough for now.